docs/pages/userdocs/workflow.dox

```diff
@@ -0,0 +1,97 @@
+ *
+ * For users wanting to integrate NWB with a particular data acquisition software, here
+ * we outline the steps for a single recording from file creation to saving.
+ *
+ * 1. Create the I/O object (e.g,.  \ref AQNWB::HDF5::HDF5IO "HDF5IO") used for 
+ *    writing data to the file on disk.
+ * 2. Create the \ref AQNWB::NWB::RecordingContainers "RecordingContainers" object 
+ *    used for managing \ref AQNWB::NWB::Container "Container" objects for storing recordings.
+ * 3. Create the \ref AQNWB::NWB::NWBFile "NWBFile" object used for managing and creating NWB 
+ *    file contents.
+ * 4. Create the \ref AQNWB::NWB::Container "Container" objects (e.g., 
+ *    \ref AQNWB::NWB::ElectricalSeries "ElectricalSeries") used for recording and add them 
+ *    to the \ref AQNWB::NWB::RecordingContainers "RecordingContainers".
+ * 5. Start the recording.
+ * 6. Write data.
+ * 7. Stop the recording and close the \ref AQNWB::NWB::NWBFile "NWBFile".
+ * 
+ * Below, we walk through these steps in more detail.
+ * 
+ * 
+ * \subsection create_io 1. Create the I/O object.
+ * 
+ * First, create an I/O object (e.g., \ref AQNWB::HDF5::HDF5IO "HDF5IO") used for writing 
+ * data to the file. AqNWB provides the convenience method, \ref AQNWB::createIO "createIO"
+ * to create this object using one of the supported backends. For more fine-grained
+ * control of different backend parameters, you can create your own `std::shared_ptr` 
+ * using any of the derived \ref AQNWB::BaseIO "BaseIO" classes.
+ * 
+ * \snippet tests/examples/testWorkflowExamples.cpp example_workflow_io_snippet
+ *
+ *
+ * \subsection create_recording_container 2. Create the RecordingContainer object.
+ * 
+ * Next, create a \ref AQNWB::NWB::RecordingContainers "RecordingContainers"  object to manage the 
+ * different \ref AQNWB::NWB::Container "Container" objects with the datasets that you would 
+ * like to write data to.
+ * 
+ * \snippet tests/examples/testWorkflowExamples.cpp example_workflow_recording_containers_snippet
+ * 
+ *
+ * \subsection create_nwbfile 3. Create the NWBFile
+ * 
+ * Next, constructs the \ref AQNWB::NWB::NWBFile "NWBFile"  object, using the I/O object as an input. 
+ * Then, initialize the object to create the basic file structure of the NWBFile.
+ *
+ * \snippet tests/examples/testWorkflowExamples.cpp example_workflow_nwbfile_snippet
+ *
+ * 
+ * \subsection create_datasets 4. Create datasets and add to RecordingContainers.
+ *
+ * Next, create the different data types (e.g. \ref AQNWB::NWB::ElectricalSeries "ElectricalSeries" 
+ * or other AQNWB::NWB::TimeSeries "TimeSeries") that you would like to write data into. After 
+ * creation, these objects are added to the  \ref AQNWB::NWB::RecordingContainers "RecordingContainers"  
+ * object so that it can manage access and data writing during the recording process. 
+ * 
+ * \snippet tests/examples/testWorkflowExamples.cpp example_workflow_datasets_snippet
+ *
+ *
```

Comment:
I would add a comment along the lines of 

*Ownership of the \ref AQNWB::NWB::ElectricalSeries "ElectricalSeries" is being transferred to the \ref AQNWB::NWB::RecordingContainers "RecordingContainers" object, so that we can
again access it from here via ....* 

and then this is also where we could show how to retrieve the object `ElectricalSeries` object again and how to use the index.